pyfranca/franca_lexer.py

```diff
@@ -118,22 +124,47 @@ def t_STRUCTURED_COMMENT(t):
         r"<\*\*(.|\n)*?\*\*>"
         t.lexer.lineno += t.value.count("\n")
 
-    # Identifier
+    # File name (quoted)
     # noinspection PyPep8Naming,PyIncorrectDocstring
     @staticmethod
-    def t_ID(t):
+    def t_FILE_NAME(t):
         # noinspection PySingleQuotedDocstring
-        r"[A-Za-z][A-Za-z0-9_]*"
-        t.type = Lexer._keyword_map.get(t.value, "ID")
+        r"\"([^\n]|\.)*?\""
+        t.value = t.value[1:-1]
         return t
 
-    # File name (quoted)
     # noinspection PyPep8Naming,PyIncorrectDocstring
     @staticmethod
-    def t_FILE_NAME(t):
+    def t_STRING_VAL(t):
         # noinspection PySingleQuotedDocstring
-        r"\"([^\n]|\.)*?\""
-        t.value = t.value[1:-1]
+        r"\"[^\"]*\""
+        t.value = t.value.replace("\"", '')
+        t.value = str(t.value)
+        return t
+
+    # noinspection PyPep8Naming,PyIncorrectDocstring
+    @staticmethod
+    def t_FLOAT_VAL(t):
+        # noinspection PySingleQuotedDocstring
+        # r"[-+]?(\d+)?(\.(\d+)?([eE][-+]?\d+)?|[eE][-+]?\d+)[f]"
+        # currently valid float string without leading digits (.043f) are not supported
+        # result in conflict with text like org.franca... matche ".f" --> need to improve regex
```

Comment:
No this is not a promblen of your grammar rule. The problem is that  the dot "." is part of the float regular expression and it is a literal. So we have to deal with the problem that we 2 valid strings that could begin with a dot and end with "d" or "f". 
if I made the digits before the dot optional in the regex and have a string like "package examples.franca". The lexer tokenize it to
PACKAGE        ID             FLOAT_VAL   ID 
package     examples              .f            ranca

I think this is a problem of the regular expression I use to match float and double values.